### ISSUE TITLE
A question about the training recipe

### ISSUE THREAD

**Issue by FiorenzoParascandolo1** [naver-ai/rope-vit#21]
Hi, thank you for the great work and for sharing the code!

In your paper, you mention:

> We train ViT with a strong supervised learning training recipe for ImageNet-1k, DeiT-III 400 epochs training recipe.

However, I noticed that in the training script provided in this repo (for example, the rope_mixed_deit_small_patch16_LS configuration), some hyperparameters differ from those in the original DeiT-III paper. Specifically:

weight decay is set to 0.03 (instead of 0.02 in DeiT-III)

learning rate is set to 4e-3 (instead of 3e-3 in DeiT-III)

Could you please clarify whether these differences are intentional or if these scripts are meant to match the exact DeiT-III recipe?

Thanks in advance for your help!

**Comment by bhheo** [comment]
Hi
I appreciate your interest in our work

Our code is exactly the same as 400 epochs training recipe of DeiT-III 
You might be confused due to the difference between DeiT-III paper's Table 1 and ours
The setting in DeiT-III paper's Table 1 is not DeiT-III's final recipe

You can find the learning rate setting in the official DeiT-III repo
https://github.com/facebookresearch/deit/blob/7e160fe43f0252d17191b71cbb5826254114ea5b/README_revenge.md?plain=1#L81
They set `lr=4e-3` only for the `ViT-small` size

For other sizes, they use `lr=3e-3` and we also use `lr=3e-3`
https://github.com/facebookresearch/deit/blob/7e160fe43f0252d17191b71cbb5826254114ea5b/README_revenge.md?plain=1#L412
https://github.com/naver-ai/rope-vit/blob/ce85c2e56c5ff3aadc10e1a849872081f7b86293/deit/README.md?plain=1#L25

Since the official codebase only includes 800 epochs setting, we have to find the weight decay setting from the paper.
For weight decay, please take a look at Appendix B of DeiT-III paper.
`We increase the weight decay from 0.02 to 0.05`
I understand this as 
| Epoch | Weight decay |
|-------|-------------------|
| 300   | 0.02              |
| 400   | 0.03              |
| 600   | 0.04              |
| 800   | 0.05              |

So, we use 0.03 weight decay for 400 epochs and successfully reproduce DeiT-III's performance
Thus, to the best of my knowledge, it is the same as DeiT-III's 400 epochs recipe

Please let me know if you have any further questions

**Comment by FiorenzoParascandolo1** [comment]
Thank you very much for your detailed and helpful response.

I still have two questions:

Do you use EMA (Exponential Moving Average) in your training procedure? From the DeiT-III paper, it seems that they do not use it. However, in that case, it's important to explicitly include the --no-model-ema flag, since otherwise it might be enabled by default in the official codebase.

From the training log of DeiT-III available online, it looks like they use the --bce-loss flag. Did you also include --bce-loss in your training?

Thanks again for your time and support!

**Comment by bhheo** [comment]
Hi

Here are answers to your questions

In general, you are correct, it is necessary to use`--no-model-ema`
But, in DeiT code, model-ema evaluation process is already excluded from the evaluation process
https://github.com/naver-ai/rope-vit/blob/ce85c2e56c5ff3aadc10e1a849872081f7b86293/deit/main.py#L436
https://github.com/naver-ai/rope-vit/blob/ce85c2e56c5ff3aadc10e1a849872081f7b86293/deit/main.py#L470
i.e. there is no evaluation with model_ema like https://github.com/huggingface/pytorch-image-models/blob/v0.4.12/train.py#L598-L603

So, regardless of `--no-model-ema`, the code only prints non-ema accuracy
You can find that DeiT commands do not use `--no-model-ema`, even though their setting is non-ema
The ema-model is included in the save file, but they don't care about this

Our model evaluation also does not use model-ema
Because we use DeiT-III setting with minimum changes, some details like not using `--no-model-ema` are the same as DeiT-III

---------

You can find `--bce-loss` in our README. 
Search `--bce-loss` at https://github.com/naver-ai/rope-vit/tree/main/deit 



I hope this clarifies your questions. Let me know if you need anything else.

**Comment by FiorenzoParascandolo1** [comment]
thanks for all clarification!
